dallinger/command_line/__init__.py

```diff
@@ -306,13 +306,23 @@ def sandbox(verbose, app, archive):
 @click.option("--verbose", is_flag=True, flag_value=True, help="Verbose mode")
 @click.option("--app", default=None, help="ID of the deployed experiment")
 @click.option("--archive", default=None, help="Optional path to an experiment archive")
+@click.option(
+    "--open-recruitment",
+    is_flag=True,
+    help="Recruitment should start automatically when the experiment launches",
+)
 @require_exp_directory
 @fail_on_unsupported_urls
 @report_idle_after(21600)
-def deploy(verbose, app, archive):
+def deploy(verbose, app, archive, open_recruitment):
     """Deploy app using Heroku to MTurk."""
     return _deploy_in_mode(
```

Comment:
Instead of hard-coding `{"mode": mode, "open_recruitment": open_recruitment}` in `_deploy_in_mode`, you should write as the first line of the `deploy` function

```py
cli_invocation_options = globals()
```

and then pass it to `deploy_in_mode`.

Reply:
You mean something like this?
```
    cli_invocation_options = globals()
    cli_invocation_options["mode"] = "live"
    cli_invocation_options["open_recruitment"] = open_recruitment

    return _deploy_in_mode(
        mode="live",
        verbose=verbose,
        log=log,
        app=app,
        archive=archive,
        cli_invocation_options=cli_invocation_options,
    )
```

Reply:
`cli_invocation_options` means the options that are present in the CLI invocation. `mode` is not present in the CLI invocation, so I would not include it. `open_recruitment` should already be part of `globals()` (because it is one of the arguments in the function call) so you shouldn't need to add it explicitly.

Reply:
After trying to implement your `cli_invocation_options = locals()` idea, I can't seem to make sense out of it. 
We have an existing `open_recruitment` param from `click` that would not be used anymore in `_deploy_in_mode`. But I need to have it declared in `_deploy_in_mode`, because we use `kwargs`. Instead we now have an additional `cli_invocation_options` param we need to pass around.
I now put `cli_invocation_options = locals()` in `_deploy_in_mode` instead. Then we only have the unused `open_recruitment` param.

Reply:
I did another round of refactoring the sandbox/deploy commands. Please have another look.

Reply:
Need to fix some tests still...

Reply:
Done fixing tests.